File: trunk/openrest4net/openrest4net-api/com/googlecode/openrest/Address.cs
﻿using System;
using Newtonsoft.Json;

namespace com.googlecode.openrest
{
    public class Address
    {
        public Address(string city, string street, string number, string apt,
            string floor, string entrance, string comment, LatLng latLng)
        {
            this.city = city;
            this.street = street;
            this.number = number;
            this.apt = apt;
            this.floor = floor;
            this.entrance = entrance;
            this.comment = comment;
            this.latLng = latLng;
        }

        /** Empty constructor required for initialization from JSON-encoded string. */
        public Address() { }

        [JsonIgnore]
        public string StreetAddress
        {
            get
            {
                return street + ' ' + number + ", " + city;
            }
        }

        public string city;
        public string street;
        public string number;
        public string apt;
        public string floor;
        public string entrance;
        public string comment;
        public LatLng latLng;
    }
}

File: trunk/openrest4j/openrest4j-api/src/com/googlecode/openrest/v1_1/Address.java
package com.googlecode.openrest.v1_1;

import java.io.Serializable;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Address implements Serializable {
    public Address(String country, String city, String street, String number,
    		String apt, String floor, String entrance, String comment, LatLng latLng) {
    	this.country = country;
        this.city = city;
        this.street = street;
        this.number = number;
        this.apt = apt;
        this.floor = floor;
        this.entrance = entrance;
        this.comment = comment;
        this.latLng = latLng;
    }

    /** Default constructor for JSON deserialization. */
    public Address() {}

    public String streetAddress() {
        return street + ' ' + number + ", " + city + ", " + country;
    }

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String country;
    
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String city;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String street;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String number;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String apt;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String floor;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String entrance;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String comment;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public LatLng latLng;
    
    private static final long serialVersionUID = 1L;
}
